compileopts/target.go

```diff
@@ -88,8 +88,17 @@ func (spec *TargetSpec) overrideProperties(child *TargetSpec) {
 			if !src.IsNil() {
 				dst.Set(src)
 			}
-		case reflect.Slice: // for slices, append the field
+		case reflect.Slice: // for slices, append the field and check for duplicates
 			dst.Set(reflect.AppendSlice(dst, src))
+			for i := 0; i < dst.Len(); i++ {
+				v := dst.Index(i).String()
+				for j := i + 1; j < dst.Len(); j++ {
+					w := dst.Index(j).String()
+					if v == w {
+						panic("duplicate value '" + v + "' in field : " + field.Name)
```

Comment:
Perhaps add info about which target failed here e.g. target name.

Reply:
No such information here?
All we have is `TargetSpec` struct, i.e. different fields, but not the actual file name.

I don't think we need this either, since we always know what target we are building or flashing. And the problem is either in the target file itself or in a target up the chain.

In this repo, smoke test  that builds every board target shall catch all such cases if we ever have them and the last target built, and failed, is the one that has the problem.

Reply:
OK fair enough. I'll go with that.